gateway/middleware.go

```diff
@@ -608,6 +606,11 @@ func (t BaseMiddleware) ApplyPolicies(session *user.SessionState) error {
 
 	// If some APIs had only ACL partitions, inherit rest from session level
 	for k, v := range rights {
+		if !didACL[k] {
+			delete(rights, k)
```

Comment:
So, this code part assumes that if there were no policies which defined ACL, we should inherit it from the session object. In old logic it was do nothing. But in new logic, you just remove it, which can broke case when policies not used or no policy with ACL?

Reply:
I started writing comment for the first line too:

>I think it may introduce a new bug, when you added Quota for specific API, but not added ACL. 
I think the proper fix will be first run this code for all ACL/non-partition policies, to understand which APIs are allowed.

But then realised that this line attempts to fix it. And on line 664 https://github.com/TykTechnologies/tyk/pull/4913/files#diff-703054910891a4db633eca0f42ed779d6b4fa75cd9b3aa4c503e681364201c1bR664 we handle case when session rights are inherited. Which also probably another bug, since we we had a session with own ACL, and policy with customer rate/quotas partitions, it will loose this per API info. But I guess we can treat it as edge case.